src/rules/no-static-element-interactions.ts

```diff
@@ -0,0 +1,54 @@
+import { Rule } from "eslint";
+import {
+  defineTemplateBodyVisitor,
+  getElementAttributeValue,
+  hasOnDirectives,
+  interactiveHandlers,
+  isHiddenFromScreenReader,
+  isInteractiveElement,
+  isInteractiveRole,
+  isPresentationRole, makeDocsURL
+} from "../utils";
+import { dom } from "aria-query";
+
+const rule: Rule.RuleModule = {
+  meta: {
+    type: "problem",
+    docs: {
+      url: makeDocsURL("no-static-element-interactions")
+    },
+    messages: {
+      default: "Visible, non-interactive elements should not have an interactive handler."
+    },
+    schema: []
+
+  },
+  create (context: Rule.RuleContext): Rule.RuleListener {
+    return defineTemplateBodyVisitor(context, {
+      VElement (node) {
+        const role = getElementAttributeValue(node, "role");
+
+        const domElements = [...dom.keys()];
+
+        if(!domElements.includes(node.name)) {
+          return;
+        }
```

Comment:
I had to stop and think about what this was doing, I find that a comment here could be of help to the next person.

```suggestion
        if(!domElements.includes(node.name)) {
          // ignore custom components/elements, only test native dom elements
          return;
        }
```

Reply:
(important) actually, more than that, there is this other function elsewhere on the project that could be of use right here. We could transform this check into another `util`, it would be great.

https://github.com/vue-a11y/eslint-plugin-vuejs-accessibility/blob/576d714bec7f114e2c89967fa2056bfeeabc1908/src/rules/click-events-have-key-events.ts#L24-L29

Reply:
Updated – added new reusable `isCustomComponent`